assets/javascripts/discourse/models/filter-rule.js.es6

```diff
@@ -1,17 +1,22 @@
 import RestModel from 'discourse/models/rest';
 
 export default RestModel.extend({
-  category_id: -1,
+  category_id: '',
   channel: '',
   filter: null,
 
   category: function() {
     var id = this.get('category_id');
 
-    if (id === "0")
-      return Discourse.Category.create({ name: 'All Categories', id: 0 });
-    else {
-      return Discourse.Category.findById(id) || { id: id, name: 'Deleted Category' };
+    switch (id) {
+      case '*':
+        return Discourse.Category.create({ name: I18n.t('slack.choose.all_categories'), id: '*' });
+        break;
+      case '':
```

Comment:
@tgxworld Now I understand your point. But the problem is currently I am not calculating that row with `AND` condition. I am doing it like `OR` condition :( :(.

I mean `{channel: "#team", category_id: "2", tags: ["dev", "bug"], filter: "follow"}` in this example Discourse will send message about a topic to `#team` channel either if it exist on category_id `2` or in tags `["dev", "bug"]`.

I just messed it. Now I have to change basic message sending code??